test-e2e/tests/payments/subscription_test.ts

```diff
@@ -157,26 +156,13 @@ function runTestSuite(props: ProviderProps, providerName: string) {
   });
 
   Scenario(`I can finish my subscription with credit card - ${providerName}`, async ({ I }) => {
-    paidLoginContext = await I.registerOrLogin(paidLoginContext);
-
-    await goToCheckout(I);
-
-    I.payWithCreditCard(
-      props.paymentFields.creditCardholder,
-      props.creditCard,
-      props.paymentFields.cardNumber,
-      props.paymentFields.expiryDate,
-      props.paymentFields.securityCode,
-      props.fieldWrapper,
-    );
-
-    await finishAndCheckSubscription(I, addYear(today), today, props.yearlyOffer.price);
+    await registerAndSubscribe(I, props, today);
 
     I.seeAll(cardInfo);
   });
 
   Scenario(`I can cancel my subscription - ${providerName}`, async ({ I }) => {
-    paidLoginContext = await I.registerOrLogin(paidLoginContext);
+    await registerAndSubscribe(I, props, today);
```

Comment:
Again, these tests have a subtle change. Before your changes it makes sure that subscription data and invoices are properly reloaded when logging back in. With your changes it just tests that it's initially populated properly when the subscription is first created, and you would have to fully clear the browser to test otherwise. I think the former is a more high traffic use case (the initial case only happens once, and the subsequent login and viewing happens every other time.) Honestly it's valuable to test both scenarios.

Maybe you can combine these last 4 scenarios into a single flow (create user, finish subscription, cancel subscription, renew subscription, view invoices.) And run it once with a single login and then a second time with logging out and clearing the browser between each step?